src/webui/www/private/scripts/client.js

```diff
@@ -487,20 +476,22 @@ window.addEvent('load', function() {
 
         const all = torrentsTable.getRowIds().length;
         let uncategorized = 0;
-        Object.each(torrentsTable.rows, function(row) {
+        for (const key in torrentsTable.rows) {
+            if (!Object.hasOwn(torrentsTable.rows, key))
+                continue;
+
+            const row = torrentsTable.rows[key];
             if (row['full_data'].category.length === 0)
                 uncategorized += 1;
-        });
+        }
         categoryList.appendChild(create_link(CATEGORIES_ALL, 'QBT_TR(All)QBT_TR[CONTEXT=CategoryFilterModel]', all));
         categoryList.appendChild(create_link(CATEGORIES_UNCATEGORIZED, 'QBT_TR(Uncategorized)QBT_TR[CONTEXT=CategoryFilterModel]', uncategorized));
 
         const sortedCategories = [];
-        Object.each(category_list, function(category) {
-            sortedCategories.push(category.name);
-        });
-        sortedCategories.sort((leftCategory, rightCategory) => {
-            const leftSegments = leftCategory.split('/');
-            const rightSegments = rightCategory.split('/');
+        category_list.forEach((category, hash) => sortedCategories.push([category.name, hash, category.torrents.size]));
```

Comment:
Wouldn't it be better if sortedCategories is array of Maps?
`categoryCount += sortedCategories[j][2]` etc. (indexed by numbers) look not so friendly.

Reply:
>Wouldn't it be better if sortedCategories is array of Maps?

Not really in terms of performance. It will incur a relative (and unnecessary) complex lookup from Map.

Reply:
It is essentially an array of some structure types. I would expect its elements to be objects or maps. Well, or at least, if using arrays significantly improves performance, then at least use named constants to access its members.

P.S. I don't care much about WebUI code so whatever you want.

Reply:
>It is essentially an array of some structure types. 

It is a tuple in C++ terms.

>then at least use named constants to access its members.

What do you mean?

Reply:
> > It is essentially an array of some structure types.
> 
> It is an array of tuples in C++ terms.

No. By "structure types" I mean `struct`, e g.:
```
struct CategoryNode
{
    QString name;
    int hash;
    int torrentsCount;
};

QList<CategoryNode> sortedCategories;

// and then:
categoryCount += sortedCategories[j].torrentsCount; 
```

Reply:
>No. By "structure types" I mean struct, e g.:

If you expect performant equivalent code exist in Javascript then there isn't any. JS `Object` type is indexed by string so there will still be a lookup. There are no `struct` equivalent in JS.

Reply:
Ok. But you can still use named constants mentioned above to make logic more readable.


> > then at least use named constants to access its members.
> 
> What do you mean?

I mean "named constants". Not sure about JS, but in C++:
```c++
const int CATEGORY_NAME = 0;
const int CATEGORY_HASH = 1;
const int CATEGORY_TORRENTS_COUNT = 2;

// OR

enum
{
    CATEGORY_NAME = 0,
    CATEGORY_HASH = 1
    CATEGORY_TORRENTS_COUNT = 2
};

// and then:

categoryCount += sortedCategories[j][CATEGORY_TORRENTS_COUNT];
```

Reply:
>But you can still use named constants mentioned above to make logic more readable.

I somehow find it redundant in the case of `categoryCount += sortedCategories[j][2]`.
Reason is that
1. You need to maintain more code (the named constants) outside the 'structure'
2. Usually there will be destructuring assignment statement (for example `const [categoryName, categoryHash] = sortedCategories[i];`) accompanied with such 'structure types' and it makes the named constants redundant/dubious.

I can declare a `const CATEGORY_TORRENTS_COUNT = 2;` and use it but another reader might wonder why there is a named constant for this exact index while other indexes have none. Of course I can declare all constants but not all of them will be referenced due to using destructuring assignment which in turn makes it redundant.

So I would prefer keeping it as is. This one imperfection can be overlooked IMO.

Reply:
I just found out that using named properties (`Object`) was actually faster than an `Array` by a non-trivial amount. 
Damn... common knowledge sometimes doesn't apply in javascript world.
PR is updated.

Reply:
> Damn... common knowledge sometimes doesn't apply in javascript world.

💯